### PR TITLE
Move package:integration_test to flutter/flutter

### PR DESCRIPTION
**Pull request by dnfield**
This will resolve issues around circular dependency management. It will also make sure that the integration_test package is tied tightly to API changes in the Flutter SDK at a specific version.

@nturgut @yjbanov @jonahwilliams

### CODE REVIEW

**Comment by dnfield** [comment]
This is going to require a large number of analysis changes. :(

**Comment by dnfield** [comment]
Windows framework_tests is a flake - filed https://github.com/flutter/flutter/issues/69922

**Comment by dnfield** [comment]
I think this is passing CI now and is ready for review.

File: packages/integration_test/.gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.DS_Store
```
Comment (jonahwilliams) [inline packages/integration_test/.gitignore:1]: I think there is a global gitignore that will handle this

File: packages/integration_test/.metadata [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# This file tracks properties of this Flutter project.
```
Comment (jonahwilliams) [inline packages/integration_test/.metadata:1]: this could be deleted

File: packages/integration_test/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
+## 0.9.2+2
```
Comment (jonahwilliams) [inline packages/integration_test/CHANGELOG.md:1]: What should we do with the version changelog now that this no longer has a version number?

File: packages/integration_test/example/.metadata [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# This file tracks properties of this Flutter project.
```
Comment (jonahwilliams) [inline packages/integration_test/example/.metadata:1]: Same here

File: packages/integration_test/CHANGELOG.md (hunk 3 above)
Comment (dnfield) [reply packages/integration_test/CHANGELOG.md:1]: I'd like to keep this here for historical purposes, but I don't think we'll keep updating the changelog. I can add a notice.

File: packages/integration_test/.metadata (hunk 2 above)
Comment (dnfield) [reply packages/integration_test/.metadata:1]: done

File: packages/integration_test/.gitignore (hunk 1 above)
Comment (dnfield) [reply packages/integration_test/.gitignore:1]: baleeted

File: packages/integration_test/example/.metadata (hunk 4 above)
Comment (dnfield) [reply packages/integration_test/example/.metadata:1]: Done

**Comment by dnfield** [comment]
Precache failure is a flake unrelated to this PR.

**Review by jonahwilliams — APPROVED** [review]
Overall LGTM for moving integration_test.

I did not closely review the code itself as it is being copied as-is for the most part

**Pull request merged**